plugins/att_cybersecurity_alienvault_otx/komand_att_cybersecurity_alienvault_otx/connection/connection.py

```diff
@@ -1,27 +1,22 @@
-import komand
+import insightconnect_plugin_runtime
 from .schema import ConnectionSchema, Input
 
 # Custom imports below
 from OTXv2 import OTXv2
+from insightconnect_plugin_runtime.exceptions import PluginException, ConnectionTestException
 
 
-class Connection(komand.Connection):
+class Connection(insightconnect_plugin_runtime.Connection):
     def __init__(self):
         super(self.__class__, self).__init__(input=ConnectionSchema())
         self.client = None
 
-    def connect(self, params):
-        api_key, otx_server = (
-            params.get(Input.API_KEY).get("secretKey", None),
-            params.get(Input.URL),
-        )
-        try:
-            self.client = OTXv2(api_key, server=otx_server)
-        except Exception as e:
-            raise Exception(f"An error has occurred while connecting: {e}")
+    def connect(self, params={}):
+        self.client = OTXv2(params.get(Input.API_KEY).get("secretKey"), server=params.get(Input.URL))
 
     def test(self):
         try:
             self.client.search_users("")
-        except Exception as e:
-            raise Exception(f"Unable to connect, error: {e}")
+            return {"success": True}
+        except Exception:
+            raise ConnectionTestException(preset=PluginException.Preset.API_KEY)
```

Comment:
Additionally we can check if `self.client` is an instance of object `OTXv2`

Reply:
If `client` is empty or there is no `search_users` method, it will throw an exception. In this situation, we will catch this exception and re-raise it.